core/iwasm/interpreter/wasm_runtime.h

```diff
@@ -112,6 +112,10 @@ struct WASMMemoryInstance {
     MemBound mem_bound_check_8bytes;
     MemBound mem_bound_check_16bytes;
 #endif
+
+#if WASM_ENABLE_SHARED_MEMORY != 0
+    korp_mutex lock;
+#endif
```

Comment:
Putting a new field here will break the AOT ABI, or, the layout of WASM/AOT module instance will change and the AOT file compiled by old wamrc won't be able to run on new wamr runtime, see:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_runtime.h#L330-L338
The offset of global data and table instance will change.

Could we add the mem_lock to `WASMSharedMemory`:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/common/wasm_shared_memory.h#L31
Seems it is more reasonable.

Reply:
Right, I can add another lock there, in addition to the already existing one https://github.com/bytecodealliance/wasm-micro-runtime/blob/1c17665f68e118cd9c6bc9b8782f1d0cda5994c9/core/iwasm/common/wasm_shared_memory.h#L24

Reply:
Actually, maybe I can use the same; the already existing one is only used to protect the `ref_count` https://github.com/bytecodealliance/wasm-micro-runtime/blob/1c17665f68e118cd9c6bc9b8782f1d0cda5994c9/core/iwasm/common/wasm_shared_memory.h#L31